### PR TITLE
chore(content-uploader): Migrate Footer

### PR DESCRIPTION
**Pull request by greg-in-a-box**
- Converted to typescript
- Converted old `.js` to `.js.flow`
- Converted to react-testing-library 
- Deleted snapshots
- Added a loading aria label for the blueprint button
- Replaced old BUIE buttons with blueprint buttons

### CODE REVIEW

File: src/elements/content-uploader/Footer.js.flow [hunk 1]
```diff
@@ -1,12 +1,6 @@
-/**
- * @flow
- * @file Footer component
- */
```
Comment (greg-in-a-box) [inline src/elements/content-uploader/Footer.js.flow:4]: apparently, we dont need this anymore if the ext is .js.flow

File: src/elements/common/messages.js [hunk 2]
```diff
@@ -1015,6 +1015,11 @@ const messages = defineMessages({
         defaultMessage: 'Box Doc Gen',
         description: 'Icon title for a Box file of type DocGen template',
     },
+    loading: {
+        id: 'be.loading',
+        description: 'Loading aria label.',
+        defaultMessage: 'loading',
```
Comment (jpan-box) [inline src/elements/common/messages.js:1021]: does this need to be capitalized? i don't think so since it's just an aria label right?

**Review by jpan-box — APPROVED** [review]

File: src/elements/common/messages.js (hunk 2 above)
Comment (greg-in-a-box) [reply src/elements/common/messages.js:1021]: I dont think they annunciate it any differently with the screen readers

File: src/elements/content-uploader/Footer.js.flow [hunk 3]
```diff
@@ -17,12 +11,13 @@ type Props = {
     hasFiles: boolean,
     isDone: boolean,
     isLoading: boolean,
-    onCancel: Function,
-    onClose?: Function,
-    onUpload: Function,
+    onCancel: any,
```
Comment (JChan106) [inline src/elements/content-uploader/Footer.js.flow:14]: Why did this change to `any`? Are we able to use arrow syntax for the types?

File: src/elements/content-uploader/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+const Footer = ({ isLoading, hasFiles, errorCode, onCancel, onClose, onUpload, fileLimit, isDone }: FooterProps) => {
+    const intl = useIntl();
+    const isCloseButtonDisabled = hasFiles;
+    const isCancelButtonDisabled = !hasFiles || isDone;
+    const isUploadButtonDisabled = !hasFiles;
+
+    let message;
+    switch (errorCode) {
```
Comment (JChan106) [inline src/elements/content-uploader/Footer.tsx:28]: Does this need to use useEffect and useState to know when errorCode changes?

File: src/elements/content-uploader/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        case ERROR_CODE_UPLOAD_FILE_LIMIT:
+            message = <FormattedMessage {...messages.uploadErrorTooManyFiles} values={{ fileLimit }} />;
+            break;
+        default:
+        // ignore
```
Comment (JChan106) [inline src/elements/content-uploader/Footer.tsx:33]: is this comment needed? If so, can we give more info

File: src/elements/content-uploader/__tests__/Footer.test.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import noop from 'lodash/noop';
+import { render, screen } from '../../../test-utils/testing-library';
+import Footer from '../Footer';
```
Comment (JChan106) [inline src/elements/content-uploader/__tests__/Footer.test.tsx:4]: Move this below  the `noop` import

File: src/elements/content-uploader/Footer.tsx (hunk 5 above)
Comment (greg-in-a-box) [reply src/elements/content-uploader/Footer.tsx:33]: probably not, left over from the conversion from js, removing

Comment (greg-in-a-box) [reply src/elements/content-uploader/Footer.tsx:33]: actually replacing the switch to an if

File: src/elements/content-uploader/__tests__/Footer.test.tsx (hunk 6 above)
Comment (greg-in-a-box) [reply src/elements/content-uploader/__tests__/Footer.test.tsx:4]: So this is a edge case, if I move the footer import above the test-utils/ import, all the test fail because, in the testing import its unmocking the intl stuff, until we can completely unmock react-intl we cant move the import above it.

If we have to move it above the testing import I have to add 
`jest.unmock('react-intl');` to the test file.

Which one do you prefer? leaving as is or move and add the unmock directly to the test file?

Comment (JChan106) [reply src/elements/content-uploader/__tests__/Footer.test.tsx:4]: Oh interesting, I have no preference, I think leaving as is is okay then

**Review by JChan106 — APPROVED** [review]
lgtm

**Review by jpan-box — APPROVED** [review]
